diff --git a/fs/coredump.c b/fs/coredump.c
index ae6b05629ca1..186f077daa3f 100644
--- a/fs/coredump.c
+++ b/fs/coredump.c
@@ -1,3 +1,4 @@
+#include <linux/sched/debug.h>
 #include <linux/slab.h>
 #include <linux/file.h>
 #include <linux/fdtable.h>
@@ -774,6 +775,12 @@ void do_coredump(const siginfo_t *siginfo)
 fail_creds:
 	put_cred(cred);
 fail:
+#ifdef CONFIG_FSP2
+	if (!core_dumped) {
+		show_regs(cprm.regs);
+		panic("coredump failed");
+	}
+#endif
 	return;
 }
 
